File: src/models/Food.ts
export type FoodSize = {
  id: number;
  name: string;
  price: number;
  weight: number;
  note: string;
  deleted: boolean;
};

export type FoodCategory = {
  id: number;
  name: string;
  image: string;
};

export enum FoodStatus {
  ACTIVE = "Active",
  DISABLE = "Disable",
}

export type Food = {
  id: number;
  name: string;
  description: string;
  images: string[];
  isDeleted: boolean;
  foodSizes: FoodSize[];
  category: FoodCategory;
  rating: number;
  tags: string[];
  status: FoodStatus;
  createdAt: Date;
  purchased: boolean;
};

File: src/server/src/main/java/com/springboot/fstore/entity/Food.java
package com.springboot.fstore.entity;

import jakarta.persistence.*;
import lombok.*;
import org.springframework.data.annotation.CreatedDate;
import org.springframework.data.jpa.domain.support.AuditingEntityListener;

import java.time.LocalDate;
import java.util.ArrayList;
import java.util.List;

@Getter
@Setter
@Builder
@AllArgsConstructor
@NoArgsConstructor

@Entity
@Table(name = "foods")
public class Food extends BaseEntity {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private int id;
    private String name;
    @Column(length = 3000, columnDefinition = "LONGTEXT")
    private String description;
    private String status;
    private Boolean isDeleted;

    @OneToMany(mappedBy = "food", cascade = CascadeType.ALL, fetch = FetchType.LAZY)
    private List<FoodSize> foodSizes;

    @ManyToOne
    @JoinColumn(name = "category_id")
    private Category category;

    @ManyToMany(cascade = CascadeType.ALL, fetch = FetchType.LAZY)
    @JoinTable(
            name = "food_image",
            joinColumns = @JoinColumn(name = "food_id"),
            inverseJoinColumns = @JoinColumn(name = "image_id")
    )
    private List<Image> images;

    @ManyToMany(cascade = CascadeType.ALL, fetch = FetchType.LAZY)
    @JoinTable(
            name = "food_tag",
            joinColumns = @JoinColumn(name = "food_id"),
            inverseJoinColumns = @JoinColumn(name = "tag_id")
    )
    private List<Tag> tags;

    @OneToMany(mappedBy = "food")
    private List<Comment> comments;
}
